website/src/main/java/com/google/sps/servlets/RefreshDataServlet.java

```diff
@@ -34,55 +34,60 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.Scanner;
+import java.util.Random;
 
 @WebServlet("/RefreshData")
-public final class RefreshDataServlet extends HttpServlet {
-
+public class RefreshDataServlet extends HttpServlet {
+  protected final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  protected final  Random r = new Random(20);
+ 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
     Query query = new Query("Restaurant").addSort("idNum", SortDirection.ASCENDING);
     PreparedQuery results = datastore.prepare(query);
+     Gson gson = new Gson();
+    ArrayList<String> restaurants = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
         long timestamp = System.currentTimeMillis();
         entity.setProperty("timestamp", timestamp);
-        int newOrderVolume = (int) entity.getProperty("openOrderVolume");
-        newOrderVolume+=getRandomChange();
+        long newOrderVolume = (long) entity.getProperty("openOrderVolume");
+        newOrderVolume+=(long)getRandomChange();
+        if(newOrderVolume>= 30){
+            newOrderVolume=26;
+        }
+        else if( newOrderVolume<0){
+            newOrderVolume =2;
+        }
         entity.setProperty("openOrderVolume", newOrderVolume);
+        String message = "orderVolume" + " : " + newOrderVolume;      
+        restaurants.add(message);
         datastore.put(entity);
     }
-    
 
-    Gson gson = new Gson();
-    ArrayList<String> restaurants = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-         int id = (int) entity.getProperty("idNUm");
-       int orderVolume = (int) entity.getProperty("openOrderVolume");
-       String message = "orderVolume" + " : " + orderVolume;      
-      restaurants.add(message); 
-      }
 
     // Send the JSON as the response
     response.setContentType("application/json;");
+
     response.getWriter().println(gson.toJson(restaurants));
   }
 
   /**
    * gets a random number between 0 and 20 for openOrderVolume
    */
-  private int getRandomChange() {
+  public int getRandomChange() {
```

Comment:
Nit this should be private or protected. depending on if you call it in the test.